src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,615 @@
+    SPELL_SUMMON_COLDWEAVE       = 46143,
+    SPELL_SUMMON_FROSTWIND       = 46382,
+    SPELL_SUMMON_HAILSTONE       = 46176,
+
+    // Ice Spear
+    SPELL_SUMMON_ICE_SPEAR_BUNNY = 46359,
+    SPELL_SUMMON_ICE_SPEAR_GO    = 46369,
+    SPELL_ICE_SPEAR_KNOCKBACK    = 46360,
+    SPELL_ICE_SPEAR_VISUAL       = 75498,
+    //SPELL_ICE_SPEAR_AURA       = 46371,
+    //SPELL_ICE_SPEAR_DELAY      = 46878,
+
+    // Slippery Floor
+    SPELL_SLIPPERY_FLOOR_AMBIENT = 46314,
+    //SPELL_SLIP                 = 45947,
+    //SPELL_YOU_SLIPPED          = 45946,
+    //SPELL_SLIPPERY_FLOOR       = 45945,
+
+    // Frozen Core
+    SPELL_SYNCH_HEALTH           = 46430,
+    SPELL_SUICIDE                = 45254,
+    //SPELL_ICE_BOMBARDMENT      = 46396,
+    //SPELL_FROZEN_CORE_HIT      = 46810,
+    SPELL_GHOST_DISGUISE         = 46786
+};
+
+enum Emotes
+{
+    EMOTE_EARTHEN_ASSAULT        = 0,
+    EMOTE_SUBMERGED              = 1,
+    EMOTE_EMERGE_SOON            = 2,
+};
+
+enum Says
+{
+    SAY_PLAYER_TEXT_1            = 0,
+    SAY_PLAYER_TEXT_2            = 1,
+    SAY_PLAYER_TEXT_3            = 2
+};
+
+enum Events
+{
+    EVENT_EMERGE                 = 1,
+    EVENT_SUBMERGE               = 2,
+    EVENT_ICE_SPEAR              = 3,
+    EVENT_COLD_SLAP              = 4,
+    EVENT_FLAMECALLER_BEAM       = 5,
+    EVENT_SUMMON_HAILSTONE       = 6,
+    EVENT_SUMMON_COLDWEAVE       = 7,
+    EVENT_SUMMON_FROSTWIND       = 8,
+    EVENT_SUMMON_SPEAR_BUNNY     = 9,
+    EVENT_EMOTE_EMERGE           = 10,
+    EVENT_EMOTE_SUBMERGE         = 11,
+    EVENT_EMOTE_EARTHEN_ATTACK   = 12,
+    EVENT_GHOST_VISUAL           = 13,
+    EVENT_REMOVE_GHOST_VISUAL    = 14,
+    EVENT_RESURFACE_SOON         = 15,
+    EVENT_EARTHEN_RING_ATTACK    = 16,
+    EVENT_ER_ATTACK_A            = 17,
+    EVENT_ER_ATTACK_B            = 18,
+    EVENT_ER_ATTACK_C            = 19
+};
+
+enum Actions
+{
+    ACTION_START_EVENT           = -2574500,
+    ACTION_AHUNE_RETREAT         = -2586500
+};
+
+enum Phases
+{
+    PHASE_ONE       = 0,
+    PHASE_TWO       = 1
+};
+
+Position const SummonPositions[] =
+{
+    { -95.523705f, -230.299057f, -1.256438f, 1.637029f },  // Ahune / Frozen-Core
+    { -135.871445f, -145.185211f, -1.851782f, 5.427049f }, // Totem A
+    { -129.179459f, -141.099503f, -1.992221f, 5.160010f }, // Totem B
+    { -122.147339f, -136.741470f, -2.285978f, 5.132535f }, // Totem C	
+    { -140.734543f, -139.580399f, -1.591881f, 5.427049f }, // Flamecaller A
+    { -134.587418f, -134.331085f, -1.855661f, 5.336747f }, // Flamecaller B
+    { -128.186356f, -130.267303f, -2.016853f, 5.460842f }, // Flamecaller C
+    { -73.4977f, -155.976f, -1.8987f, 3.23816f }           // Slippery Floor
+
+};
+
+class boss_ahune : public CreatureScript
+{
+public:
+    boss_ahune() : CreatureScript("boss_ahune") { }
+
+    struct boss_ahuneAI : public BossAI
+    {
+        boss_ahuneAI(Creature* creature) : BossAI(creature, NPC_AHUNE), _summons(me) { }
```

Comment:
This is wrong, the constructor asks for the "boss id" in the given instance, not the creature entry, also, SummonList is already inherited by BossAI